src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -2150,12 +2174,13 @@ def _get_network_plugin(self, enable_validation: bool = False) -> Union[str, Non
             ) = self._get_pod_cidr_and_service_cidr_and_dns_service_ip_and_docker_bridge_address_and_network_policy(
                 enable_validation=False
             )
+            network_plugin_mode = self._get_network_plugin_mode(enable_validation=False)
             if network_plugin:
-                if network_plugin == "azure" and pod_cidr:
-                    logger.warning(
-                        'When using `--network-plugin azure` without the preview feature '
-                        '`--network-plugin-mode overlay` the provided pod CIDR "%s" will be '
-                        'overwritten with the subnet CIDR.', pod_cidr
+                if network_plugin == "azure" and pod_cidr and network_plugin_mode != "overlay":
+                    raise InvalidArgumentValueError(
+                        "Please specify network plugin mode `overlay` when using pod_cidr or "
+                        "use network plugin `kubenet`. For more information about Azure CNI "
+                        "Overlay please see https://aka.ms/aks/azure-cni-overlay"
                     )
```

Comment:
Will this cause a breaking change for the customers due to the warning becoming an error?

Reply:
in theory it could but originally the validation was to raise this error and it was changed in this PR when we were adding network-plugin-mode to preview: https://github.com/Azure/azure-cli/pull/22531/files

but using --network-plugin azure with --pod-cidr is not a supported configuration in AKS RP unless you specify --network-plugin-mode overlay

so this change is really reverting it back to the original error but allowing it if using overlay

Reply:
I personally suggest that we first keep it as a warning log to avoid causing a sudden breaking change to the customers, and inform the customer that this situation will become an error in the future. Then we can turn it into the error in the next breaking change release (Build Sprint), what do you think of it?

Reply:
yeah I'm fine with that. changed back to a warning in the last update.